nntrainer/layers/common_properties.h

```diff
@@ -1586,6 +1586,65 @@ class PropsUserData final : public Property<void *> {
   using prop_tag = ptr_prop_tag;
 };
 
+/**
+ * @brief Enumeration of Tensor Life Spacn
+ */
+struct TensorLifeInfo {
+  using Enum = nntrainer::TensorLifespan;
+  static constexpr std::initializer_list<Enum> EnumList = {
+    Enum::UNMANAGED,
+    Enum::FORWARD_FUNC_LIFESPAN,
+    Enum::CALC_DERIV_LIFESPAN,
+    Enum::CALC_GRAD_LIFESPAN,
+    Enum::CALC_AGRAD_LIFESPAN,
+    Enum::CALC_GRAD_DERIV_LIFESPAN,
+    Enum::CALC_GRAD_DERIV_AGRAD_LIFESPAN,
+    Enum::FORWARD_GRAD_LIFESPAN,
+    Enum::FORWARD_GRAD_AGRAD_LIFESPAN,
+    Enum::FORWARD_DERIV_LIFESPAN,
+    Enum::BACKWARD_FUNC_LIFESPAN,
+    Enum::CALC_GRAD_DERIV_AGRAD_LIFESPAN,
+    Enum::ITERATION_LIFESPAN,
+    Enum::EPOCH_LIFESPAN,
+    Enum::FORWARD_INFER_LIFESPAN,
+    Enum::MAX_LIFESPAN};
+
+  static constexpr const char *EnumStr[] = {"unmanaged",
+                                            "forward",
+                                            "deriv",
+                                            "grad",
+                                            "agrad",
+                                            "grad_deriv",
+                                            "grad_deriv_agrad",
+                                            "forward_grad",
+                                            "forward_grad_agrad",
+                                            "forward_deriv",
+                                            "backward",
+                                            "grad_deriv_agrad",
+                                            "iteration",
+                                            "epoch",
+                                            "forward_infer",
+                                            "max"};
+};
+
+/**
+ * @brief TensorLifeSpacn, return TensorLifespan
+ *
+ */
+class TensorLife : public EnumProperty<TensorLifeInfo> {
```

Comment:
Makes me wonder if TensorLife property is something that is temporary?

Seems like (may be wrong) something we could/should infer from network graph during planning rather than to be supplied by user.

Also bit worried if it won't be to much of a knowledge for user about what is best memory managment strategy.